grails-app/controllers/com/unifina/controller/api/KeyApiController.groovy

```diff
@@ -60,9 +60,15 @@ class KeyApiController {
 
 				List keys = permissionsByKey.collect { key, permissions ->
 					Map map = key.toMap()
-					map["permission"] = permissions.find {
-						it.operation == Permission.Operation.WRITE
-					}?.operation?.id ?: Permission.Operation.READ.id
+					if (Stream.isAssignableFrom(params.resourceClass)) {
+						map["permission"] = permissions.find {
+							it.operation == Permission.Operation.STREAM_EDIT
```

Comment:
I have an idea how this whole mess could become much cleaner, but it's IMO it's not worth doing because we'll get rid of these "anonymous" (not associated with a `SecUser`) `Key`s soon anyway.

Anyway, we need to use `STREAM_PUBLISH` here.